walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -69,12 +69,13 @@
       return slotsForMandatory;
     }
 
-    /* compute the intersection between the two sets of slots */
-    Collection<TimeRange> intersection = intersection(slotsForOptional, slotsForMandatory);
-    if (intersection.isEmpty()) {
+    /* compute the slots that accommodate both optional and mandatory attendees */
+    Collection<TimeRange> overallAvailableSlots =
+        overallAvailableSlots(slotsForOptional, slotsForMandatory, request.getDuration());
```

Comment:
This code overall works, so no need to change it :) I think another way of doing it would be:
```
Collection<Event> eventClashesAllAttendees = eventsThatClashWithRequest(events, allAttendees);
Collection<TimeRange> possibleSlotsForAll = possibleSlots(eventClashesAllAttendees, request.getDuration());

return possibleSlotsForAll.isEmpty() ? possibleSlotsForMandatory : possibleSlotsForAll;
```
where `allAttendees` contains the union of mandatory and optional attendees.

This avoids having to write the extra code.